app/main.py

```diff
@@ -2,5 +2,23 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args, **kwargs) -> Callable:
+        try:
+            key = (args, frozenset(kwargs.items()))  # Незмінний ключ
```

Comment:
The use of `frozenset(kwargs.items())` is correct for creating an immutable key from the `kwargs` dictionary. However, ensure that all values in `kwargs` are themselves immutable. If any value is mutable, this will raise a `TypeError` when trying to create a `frozenset`.